src/web3/MassaAmount.ts

```diff
@@ -0,0 +1,78 @@
+import BigNumber from 'bignumber.js';
+
+export enum MASSA_UNIT {
+  MASSA,
+  MILLI_MASSA, // 10-3
+  MICRO_MASSA, // 10-6
+  NANO_MASSA, // 10-9
+}
+
+const MILLI = 3;
+const MICRO = 6;
+const NANO = 9;
+
+export class MassaAmount {
+  private value: BigNumber;
+  private unit: MASSA_UNIT;
+
+  constructor(value: BigNumber | number | string, unit: MASSA_UNIT) {
+    switch (unit) {
+      case MASSA_UNIT.MASSA:
+        this.value = new BigNumber(new BigNumber(value).toFixed(NANO));
+        break;
+      case MASSA_UNIT.MILLI_MASSA:
+        this.value = new BigNumber(new BigNumber(value).toFixed(MICRO));
+        break;
+      case MASSA_UNIT.MICRO_MASSA:
+        this.value = new BigNumber(new BigNumber(value).toFixed(MILLI));
+        break;
+      case MASSA_UNIT.NANO_MASSA:
+        this.value = new BigNumber(new BigNumber(value).toFixed(0));
+        break;
+    }
+    this.unit = unit;
+  }
+
+  public getValue(): BigNumber {
+    return this.value;
+  }
+
+  public getUnit(): MASSA_UNIT {
+    return this.unit;
+  }
+
+  public toRolls(): MassaAmount {
+    const valueInMassa = this.toUnit(MASSA_UNIT.MASSA);
+    return new MassaAmount(valueInMassa.value.toFixed(0), MASSA_UNIT.MASSA);
```

Comment:
`return this.toUnit(MASSA_UNIT.MASSA);` it is weird to initialize a new massaAmount object here..

Reply:
@peterjah Same argumentation as above: this is needed because of .toFixed(0)